src/test/java/com/github/martapietka/ppm2pbmSpring/Ppm2PbmControllerTest.java

```diff
@@ -2,41 +2,37 @@
 
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.test.web.server.LocalServerPort;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
+
+import org.springframework.http.MediaType;
 
 import java.nio.file.Files;
 import java.nio.file.Paths;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
 class Ppm2PbmControllerTest {
 
-    @LocalServerPort
-    private int port;
-
     @Autowired
-    private TestRestTemplate restTemplate;
+    private MockMvc mockMvc;
 
     @Test
     public void shouldReturnCorrectImage() throws Exception {
 
         String imageName = "3x3.ppm";
         byte[] expectedBytes = Files.readAllBytes(Paths.get("src/test/resources/" + imageName));
```

Comment:
To jest OK :) ale ogólnie to ten test działa trochę na słowo honoru - obecna implementacja serwisu wymaga, żebyś miała na dysku w katalogu odpowiedni plik, identyczny jak ten. Test kontrolera nie powinien od tego zależeć.

Użyj `@MockBean` i by zamockować warstwę serwisową tak, by zwracała te `expectedBytes`.